Add aws_access_key_id and aws_secret_access_key parameters

nicklewis opened the pull request:
Previously, the only ways to provide credentials to the
resolve_reference task were to put the credentials in a file or in an
environment variable. Both of those require managing some aspect of the
environment external to the task. To avoid that requirement, the task
now also accepts these credentials directly as parameters. If both
a `credentials` file and one of the individual credential parameters are
supplied, the contents of the credentials file will take precedence.

nicklewis committed:
Add aws_access_key_id and aws_secret_access_key parameters

Previously, the only ways to provide credentials to the
resolve_reference task were to put the credentials in a file or in an
environment variable. Both of those require managing some aspect of the
environment external to the task. To avoid that requirement, the task
now also accepts these credentials directly as parameters. If both
a `credentials` file and one of the individual credential parameters are
supplied, the contents of the credentials file will take precedence.

beechtom commented on tasks/resolve_reference.json:
```diff
@@ -17,6 +17,13 @@
     "credentials": {
       "type": "Optional[String]"
     },
+    "access_key_id": {
+      "type": "Optional[String]"
+    },
+    "secret_access_key": {
+      "type": "Optional[String]",
+      "sensitive": true
+    },
```
Can you add these to the [README](https://github.com/puppetlabs/puppetlabs-aws_inventory/blob/main/README.md#usage) as available options?

beechtom reviewed (approved):
👍  lgtm

nicklewis commented:
Updated the readme and renamed the parameters to be prefixed with `aws_` to match the environment variables and the keys in the credentials file. I also replaced the AWS credentials that were being used in tests with fake strings (the credentials were deactivated and deleted from my account, but were going to set off lots of security scanners...)

donoghuc commented on tasks/resolve_reference.json:
```diff
@@ -17,6 +17,13 @@
       "type": "Optional[String]"
     },
+    "aws_access_key_id": {
+      "type": "Optional[String]"
```
I think i see why this is not "sensitive" but your test case makes me hesitant https://github.com/puppetlabs/puppetlabs-aws_inventory/pull/17/files#diff-c7639b8a584e34dc59a315a1f3b5d23034c8a64db7415137a76c525bae92f89bR106 given you call it "secret" there?

nicklewis replied:
That's the other field. `aws_access_key_id` is not sensitive, `aws_secret_access_key` is sensitive.
